src/components/Drawer/Drawer.component.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { DisclosureProps, Disclosure, DisclosureContent, useDisclosureState } from 'reakit';
+
+import * as S from './Drawer.style';
+
+export type DrawerProps = DisclosureProps & {
+	disclosure: React.ComponentPropsWithRef<any>;
```

Comment:
```suggestion
	disclosure: React.ComponentPropsWithRef<HTMLButtonElement>;
```
Or maybe there too we may have trouble with some styled-components version of a button?